api_end_end_testing/src/test/java/tds/user/api/tests/CreateUserTest.java

```diff
@@ -0,0 +1,247 @@
+            .when()
+            .post("/auth/oauth2/access_token")
+            .then()
+            .statusCode(200)
+            .extract()
+            .path("access_token");
+    }
+
+    /*
+        Create user
+        Testing HTTP POST of /rest/external/user, 201 success item created
+    */
+    @Test(dependsOnMethods = "getAuthenticationCode")
+    public void createUser() {
+        email = "betsy.ross@example.com";
+
+        RestAssured.baseURI = userUri;
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        UserInfo userInfo = new UserInfo(email, "Betsy", "Ross", "800-332-4747", roleAssociations);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = null;
+
+        try {
+            jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(userInfo);
+            System.out.println("jsonInString: " + jsonInString);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post("/rest/external/user")
+            .then()
+            .statusCode(201);
+    }
+
+    /*
+        Update user
+        Testing HTTP POST of /rest/external/user, 204 success item updated
+    */
+    @Test(dependsOnMethods = "createUser")
+    public void updateUser() {
+
+        RestAssured.baseURI = userUri;
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        UserInfo userInfo = new UserInfo(email, "Betsy", "Carson", "800-332-4747", roleAssociations);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = null;
+
+        try {
+            jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(userInfo);
+            System.out.println("jsonInString: " + jsonInString);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post("/rest/external/user")
+            .then()
+            .statusCode(204);
+    }
+
+    /*
+        Create invalid user
+        Testing HTTP POST of /rest/external/user, 400 bad request
+    */
+    @Test(dependsOnMethods = "updateUser")
```

Comment:
Maybe it's just me, but I'm finding the `dependsOnMethods` to be a bit hard to follow.  I understand that the tests might need to be executed in a particular order, and that JUnit doesn't guarantee execution order (by design).  As complexity of workflows increases, I think this could become brittle and/or hard to follow.  Also, doesn't this make it a bit harder to run the tests individually?  For instance, what happens if I try to run the `updateUser()` test?  Will the `dependsOnMethods` attribute cause the dependent test to be executed?